src/main/java/io/github/oliviercailloux/jconfs/calendar/CalendarOnline.java

```diff
@@ -88,13 +89,19 @@ public CalendarOnline(CalDavCalendarGeneric connector) {
 		GenerateQuery searchQuery = new GenerateQuery();
 		CalendarQuery calendarQuery = searchQuery.generate();
 		Set<Conference> listConferencesUser = new LinkedHashSet<>();
-		List<Calendar> calendarsResult = connector.collectionCalendarsOnline.queryCalendars(this.connector.httpclient, calendarQuery);
+		List<Calendar> calendarsResult = this.connector.collectionCalendarsOnline
+				.queryCalendars(this.connector.httpclient, calendarQuery);
+
 		for (Calendar calendar : calendarsResult) {
-			ComponentList<VEvent> componentList = calendar.getComponents(Component.VEVENT);
-			Iterator<VEvent> eventIterator = componentList.iterator();
-			while (eventIterator.hasNext()) {
-				VEvent vEventFound = eventIterator.next();
-				listConferencesUser.add(ConferenceReader.createConference(vEventFound));
+			try {
+				ComponentList<VEvent> componentList = calendar.getComponents(Component.VEVENT);
+				Iterator<VEvent> eventIterator = componentList.iterator();
+				while (eventIterator.hasNext()) {
+					VEvent vEventFound = eventIterator.next();
+					listConferencesUser.add(ConferenceReader.createConference(vEventFound));
+				}
+			} catch (Exception e) {
+				System.out.println("Conference unreadable");
```

Comment:
Il me semble avoir beaucoup, beaucoup insisté en cours sur l’importance de ne pas attraper les exceptions sans rien en faire.

Reply:
Pour ce point-ci, c'est une extension du #9. Lorsque les conférences sont parcourus, si le vEvent ne contient pas le champ URL, on a une exception qui se lance. Nativement lorsque l'on crée un événement via l'interface et bien l'application plante. Nous attrapons donc l'exception tout en affichant que cette conférence n'est pas lisible

Reply:
La contrainte de l'URL reste intéressante dans le cadre d'une conférence. C'est pourquoi nous sommes partis sur cette solution d'attraper l'exception et d'afficher à l'utilisateur que l'une de ses conférences est illisible. Ce point aurait pu nous aider car cette fonction plantait auparavant car sur le calendrier de l'utilisateur il y'a un événement crée d'une mauvaise manière qui était plus ancien que nos nouveaux événements qui était eux bien crées.

Reply:
Je ne comprends pas votre réponse. Le code doit échouer rapidement s’il n’est pas capable d’accomplir sa mission, et pas se poursuivre comme si de rien n’était. Afficher grâce à System.out.println ne résoud pas le problème, comme je l’ai expliqué en cours en détail. Par ailleurs, le commentaire « Maybe add a try catch in line 97 because if an event don't have an url in vEvent it block the program. » n’est pas approprié.

Reply:
D'accord, nous trouvions qu'échouer pour une seul conférence qui n'est pas lisible n'est pas forcément une bonne idée mais si vous le souhaitez nous pouvons retirer cet ajout. Il s'avère que le lecteur ne lit que les conférences créées par notre application et peut avoir du mal avec les événements crées avec l'interface web car elles ne possèdent pas la propriété d'Url dans leur vEvent.

Reply:
Il y a d’autres alternatives que "laisser le code tel quel" (avec donc les problèmes relevés) ou "retirer la modification et planter sans recours à la moindre entrée avec url vide" (ce qui, je vous l’accorde, n’est pas terrible non plus). Je vous laisse y réfléchir et proposer une meilleure solution.

Reply:
Nous avons supprimés les try catch des deux cotés et cela fonctionne.

Reply:
Pouvez-vous m’expliquer en deux mots comment ce problème a été résolu ?

Reply:
Nous retournons la valeur nulle dans le cadre de conférence inappropriée. Une issue #34 a été créée.